synapse/handlers/oidc.py

```diff
@@ -220,11 +220,11 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
                 session, state
             )
         except (MacaroonInitException, MacaroonDeserializationException, KeyError) as e:
-            logger.exception("Invalid session for OIDC callback")
```

Comment:
I think the main source here are users initiating multiple SSO sessions at once, so probably fine to ignore

Reply:
Looking at sentry, this one is really weird. It happens when `session == b""`, which might be a result of the cookie clearing done before. Testing it a bit, on Safari, it looks like the cookie clearing line 201 does not clear the cookie properly, and instead sets it to a blank value, hence the issue.

So, it's a legit exception, and probably a legit bug because we're not clearing the cookie properly.

Reply:
Thanks, this is really insightful. I'll open a new issue referencing this and remove this change from the PR.

Reply:
That's done in 016fc20f7.